File: hw/maisincho/u3/HW17MazeSolid/MazeJava/src/main/java/ec/edu/espe/mazegenerator/controller/AsciiMazePrinter.java

package ec.edu.espe.mazegenerator.controller;

import ec.edu.espe.mazegenerator.model.Maze;
import ec.edu.espe.mazegenerator.model.Room;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class AsciiMazePrinter implements MazePrinter{
     @Override
    public void print(Maze maze) {
        Room[][] rooms = maze.getRooms();
        int width = rooms.length;
        int height = rooms[0].length;

        for (int y = 0; y < height; y++) {
            final int finalY = y; 
            for (int x = 0; x < width; x++) {
                final int finalX = x; 
                System.out.print("+");
                if (rooms[finalX][finalY].getDoors().stream().anyMatch(door -> door.getRoom1().getY() == finalY && door.getRoom2().getY() == finalY + 1)) {
                    System.out.print("   ");
                } else {
                    System.out.print("---");
                }
            }
            System.out.println("+");

            for (int x = 0; x < width; x++) {
                final int finalX = x; 
                if (rooms[finalX][finalY].getDoors().stream().anyMatch(door -> door.getRoom1().getX() == finalX && door.getRoom2().getX() == finalX + 1)) {
                    System.out.print(" ");
                } else {
                    System.out.print("|");
                }
                System.out.print("   ");
            }
            System.out.println("|");
        }

        for (int x = 0; x < width; x++) {
            System.out.print("+---");
        }
        System.out.println("+");
    }
}

File: hw/maisincho/u3/HW17MazeSolid/MazePython/controller/asciiMazePrinter.py
from controller.mazePrinter import MazePrinter
from model.maze import Maze

class AsciiMazePrinter(MazePrinter):
    def print(self, maze: Maze) -> None:
        rooms = maze.rooms
        width = len(rooms)
        height = len(rooms[0])

        for y in range(height):
            for x in range(width):
                print("+", end="")
                # Check if the door is down
                if any((door.room1.y == y and door.room2.y == y + 1) or (door.room2.y == y and door.room1.y == y + 1) for door in rooms[x][y].doors):
                    print("   ", end="")
                else:
                    print("---", end="")
            print("+")

            for x in range(width):
                # Check if there is a door on the right
                if any((door.room1.x == x and door.room2.x == x + 1) or (door.room2.x == x and door.room1.x == x + 1) for door in rooms[x][y].doors):
                    print(" ", end="")
                else:
                    print("|", end="")
                print("   ", end="")
            print("|")

        for x in range(width):
            print("+---", end="")
        print("+")